hoomd/md/EvaluatorPairMorse.h

```diff
@@ -45,15 +45,57 @@ class EvaluatorPairMorse
     {
     public:
         //! Define the parameter type used by this pair potential evaluator
-        typedef Scalar4 param_type;
+        struct param_type
+            {
+            Scalar D0;
+            Scalar alpha;
+            Scalar r0;
+
+            #ifdef ENABLE_HIP
+            // CUDA memory hints
+            void set_memory_hints() const {}
+            #endif
+
+            #ifndef __HIPCC__
+            param_type() {D0=1; alpha=0; r0=0;}
+
+            param_type(pybind11::dict v)
+                {
+                D0 = v["D0"].cast<Scalar>();
+                alpha = v["alpha"].cast<Scalar>();
+                r0 = v["r0"].cast<Scalar>();
+                }
+
+            param_type(Scalar d, Scalar a, Scalar r)
+                {
+                D0 = d;
+                alpha = a;
+                r0 = r;
+                }
+
+            pybind11::dict asDict()
+                {
+                pybind11::dict v;
+                v["D0"] = D0;
+                v["alpha"] = alpha;
+                v["r0"] = r0;
+                return v;
+                }
+            #endif
+            }
+            #ifdef SINGLE_PRECISION
```

Comment:
This struct has 3 Scalars so use 16-byte alignment in both single and double precision.